grammars/ruby.cson

```diff
@@ -60,60 +60,88 @@
 '''
 'patterns': [
   {
+    'begin': '\\bclass\\b'
     'captures':
-      '1':
+      '0':
         'name': 'keyword.control.class.ruby'
-      '2':
-        'name': 'entity.name.type.class.ruby'
-      '4':
-        'name': 'entity.other.inherited-class.ruby'
-      '5':
-        'name': 'punctuation.separator.inheritance.ruby'
-      '6':
-        'name': 'variable.other.object.ruby'
-      '7':
-        'name': 'punctuation.definition.variable.ruby'
-    'match': '''(?x)
-      ^\\s*(class)\\s+
-      (
-        (
-          [.a-zA-Z0-9_:]+
-          (\\s*(<)\\s*[.a-zA-Z0-9_:]+)?   # class A < B
-        )
-        |
-        ((<<)\\s*[.a-zA-Z0-9_:]+)         # class << C
-      )
-    '''
+    'end': '\\s*$|(?![\\s\\w.:<])'
     'name': 'meta.class.ruby'
+    'patterns': [
+      {
+        'match': '[A-Z]\\w*'
+        'name': 'entity.name.type.class.ruby'
+      }
+      {
+        'include': '#separators'
+      }
+      {
+        'begin': '<<'
```

Comment:
Seems like you could redo this as `(<<)\\s*` and directly add a contentName here to get rid of the inner pattern.